js/componentNameView.js

```diff
@@ -1,83 +1,56 @@
-         */
-        resetQuestionOnRevisit: function(type) {},
-
-        /**
-         * Used by question components instead of preRender
-         */
-        setupQuestion: function() {},
-
-        /**
-         * Used by question components instead of postRender
-         */
-
-        onQuestionRendered: function() {},
-
-        /**
-         * Called when the user clicks submit and this.model.canSubmit() returns false
-         * Not necessary but you
-         */
-        onCannotSubmit: function() {},
-
-        /**
-         * This is important and should give the user visual feedback on how they answered the question
-         * Normally done through ticks and crosses by adding classes
-         */
-        showMarking: function() {},
-
-        /**
-         * Resets the look and feel of the component
-         * This could also include resetting item data
-         * This is triggered when the reset button is clicked so it shouldn't be a full reset
-         */
-        resetQuestion: function() {},
-
-        /**
-         * Displays the correct answer to the user
-         */
-        showCorrectAnswer: function() {},
-
-        /**
-         * Displays the user's answer and hides the correct answer
-         * Should use the values stored in this.storeUserAnswer
-         */
-        hideCorrectAnswer: function() {},
-
-        /**
-         * Used by adapt-contrib-spoor to get the user's answers in the format
-         * required by the cmi.interactions.n.student_response data field
-         * @returns {string} a string representation of the user's answer
-         */
-        getResponse:function() {},
-
-        /**
-         * Used by adapt-contrib-spoor to get the type of this question in the
-         * format required by the cmi.interactions.n.type data field.
-         * Please note the answer will not store correctly unless this function returns a valid string
-         * @returns {string} one of the following: choice, matching, numeric, fill-in
-         */
-        getResponseType:function() {}
-
-    });
-
-    return ComponentNameView;
-
-});
+import QuestionView from 'core/js/views/questionView';
+
+export default class ComponentNameView extends QuestionView {
+
+  /**
+   * Used to disable the question during submit and complete stages
+   */
+  disableQuestion() {}
+
+  /**
+   * Used to enable the question during submit and complete stages
+   */
+  enableQuestion() {}
+
+  /**
+   * Used by question components instead of preRender
+   */
+  setupQuestion() {}
+
+  /**
+   * Used by question components instead of postRender
+   */
+
+  onQuestionRendered() {}
+
+  /**
+   * Called when the user clicks submit and this.model.canSubmit() returns false
+   * Not necessary but you
+   */
+  onCannotSubmit() {}
+
+  /**
+   * This is important and should give the user visual feedback on how they answered the question
+   * Normally done through ticks and crosses by adding classes
+   */
+  showMarking() {}
```

Comment:
With jsx this isn't necessary as [`isInteractive() && _canShowMarking`](https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/questionModel.js#L282) determine the show/hide of marking in the template.

[MCQ example](https://github.com/adaptlearning/adapt-contrib-mcq/blob/master/templates/mcq.jsx#L26)

Perhaps we should make that into a model property called `shouldShowMarking`? Woud make sense to have that as core behaviour.

Or at least include the mcq example in the template here.

What do you think?

Reply:
I have created a ticket to add the model property `shouldShowMarking` here https://github.com/adaptlearning/adapt-contrib-core/issues/32